orte/mca/plm/slurm/plm_slurm_module.c

```diff
@@ -277,8 +277,17 @@ static void launch_daemons(int fd, short args, void *cbdata)
      * This will *not* release the orteds from any cpu-set
      * constraint, but will ensure it doesn't get
      * bound to only one processor
-     */
-    opal_argv_append(&argc, &argv, "--cpu_bind=none");
+     *
+     * NOTE: We used to pass --cpu_bind=none on the command line.  But
+     * SLURM 19 changed this to --cpu-bind.  There is no easy way to
+     * test at run time which of these two parameters are used (see
+     * https://github.com/open-mpi/ompi/pull/6654) -- there was
+     * discussion of using --test-only to see which one works, but
+     * --test-only is only effective if you're not already inside a
+     * SLURM allocation.  Instead, set the env var SLURM_CPU_BIND to
+     * "none", which should do the same thing as --cpu*bind.
+    */
+    opal_setenv("SLURM_CPU_BIND", "none", true, &orte_launch_environ);
```

Comment:
You really should move this down to line 422 (pre-PR) and add that envar to the srun "env" environment, not the orte_launch_environ, as that is what we pass to our client procs. I don't _think_ it will hurt anything for the clients, but no reason to take a chance.

Reply:
Will do!